File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxProduct = function(nums) {
  const fst = Math.max(...nums)
  nums.splice(nums.indexOf(fst), 1)
  const snd = Math.max(...nums)
  return (fst - 1) * (snd - 1)
};

File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.go
func maxProduct(nums []int) int {
  var m1, m2 int
  for _, v := range nums {
    if v > m1 {
      m1, m2 = v, m1
    } else if v > m2 {
      m2 = v
    }
  }  
  return (m1-1)*(m2-1)
}